chain/contractWrappers.go

```diff
@@ -210,12 +210,22 @@ func (ph *VotingHandle) NewProcessTxArgs(ctx context.Context, pid [types.Process
 			return nil, fmt.Errorf("invalid storage root, root fetched and provided must be the same. Got: %x expected: %x", fetchedRoot, processData.CensusRoot)
 		}
 		// get index slot from the token storage proof contract
-		islot, err := ph.TokenStorageProof.GetBalanceMappingPosition(&ethbind.CallOpts{Context: ctx}, processMeta.EntityAddress)
+		islot, err := ph.GetTokenBalanceMappingPosition(ctx, processMeta.EntityAddressOwner[0])
 		if err != nil {
 			return nil, fmt.Errorf("cannot get balance mapping position from the contract: %w", err)
 		}
 		iSlot32 := uint32(islot.Uint64())
 		processData.EthIndexSlot = &iSlot32
+		// decode owner
+		if processData.Owner, err = hex.DecodeString(util.TrimHex(processMeta.EntityAddressOwner[1].String())); err != nil {
+			return nil, fmt.Errorf("error decoding owner address: %w", err)
+		}
+		// save sourceBlockHeight
+		sourceBlockHeight64 := uint64(processMeta.SourceBlockHeight.Uint64())
+		if sourceBlockHeight64 == 0 {
```

Comment:
Do we want to obligate the usage of sourceBlockHeight? I am not sure, I think that should be optional (as it is the field on protobuf).

Reply:
Is optional on the protobuf because process can have on-chain / off-chain census. The code here only applies to processes that need IndexSlot (on-chain census) so I assume these processes will also require to have the sourceBlockHeight stored (if we really want to have it stored)